spring-integration-core/src/main/java/org/springframework/integration/support/locks/CustomTtlLock.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.support.locks;
+
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.locks.Lock;
+
+/**
+ * A {@link Lock} implementing this interface supports the spring distributed locks with custom time-to-live value per lock
+ *
+ * @author Eddie Cho
+ *
+ * @since 6.3
+ */
+public interface CustomTtlLock extends Lock {
```

Comment:
In this PR, the `CustomTtlLock` interface is only implemented by `RedisLock` and `JdbckLock`.
It can be renamed to `DistributedLock` in the future if all the implementations of Spring distributed lock have implemented the interface.